keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -200,17 +213,138 @@ def backbone_cls(cls):
     def preprocessor_cls(cls):
         return GPT2CausalLMPreprocessor
 
-    def _get_token_probability(self, prompt, mask):
+    def call_with_cache(self, inputs, cache, cache_index=None):
+        """Forward pass of `GPT2CausalLM` with cache.
+
+        The difference between `call_with_cache` and normal `__call__` is in
+        this method, a `cache` arg is set, and the inputs is of
+        `sequence_length=1`. By cachine the previous key/value in multi-head
+        attention, we avoid recomputing the outputs of seen tokens.
```

Comment:
The docstrings might need a round of copyedits in general

Reply:
applied